ctc_lib/container_tool_command_base.go

```diff
@@ -60,12 +63,19 @@ func (ctb *ContainerToolCommandBase) AddCommand(command CLIInterface) {
 
 func (ctb *ContainerToolCommandBase) AddFlags() {
 	// Add template Flag
-	ctb.PersistentFlags().StringVarP(&flags.TemplateString, "template", "t", emptyTemplate, "Output format")
-	ctb.PersistentFlags().VarP(types.NewLogLevel(defaultLogLevel, &flags.LogLevel), "loglevel", "l", "LogLevel")
+	ctb.PersistentFlags().StringVarP(&flags.TemplateString, "template", "t", constants.EmptyTemplate, "Output format")
+	ctb.PersistentFlags().VarP(types.NewLogLevel(constants.DefaultLogLevel, &flags.LogLevel), "logLevel", "l", "LogLevel")
+	ctb.PersistentFlags().BoolVarP(&flags.UpdateCheck, "updateCheck", "u", true, "Run Update Check") // TODO Add Update Check logic
+	viper.BindPFlag("updateCheck", ctb.PersistentFlags().Lookup("updateCheck"))
+	// Also Log to StdOut
+	ctb.PersistentFlags().BoolVar(&flags.AlsoLogToStdOut, "alsoLogToStdOut", false, "Also Log to Std Out")
+
+	ctb.PersistentFlags().StringVar(&flags.LogDir, "logDir", "/tmp/", "LogDir")
```

Comment:
Is this something we could make tools specify, and not log to a file if they don't? I don't think logs in /tmp without some kind of tool specific prefix will be too easy to find.

Reply:
yeah. i want a reasonable default for logs with something you can have access too in all environments.
The destination is up for discussion.
The idea for logging to dir was to always have logs handy..  
The reason why, i want this not to be tool config but a command line flag with reasonable default is for users to see the logs for later when things go wrong and not have the need to re-run the command with --alsoLogToStdOut.
Would it make more sense if i can print a message if Command exits with an error.
Eg
```
bazel run demo/ctc:ctc_demo --  panic
See logs at /tmp/echo for more details
+ERROR: Non-zero return code '1' from command: Process exited with status 1
```
In future, we can also add a command to open a bug which will pick the log and attach it to a github issue.

Other idea i had was to put the logs in ~/<.tool>/logs like gcloud does.
In future, when we bundle all our tools with container tool box, we can override the logging dir to be ~/container_tools/<.tool>/logs

Reply:
Yeah my concern with just /tmp is that it would have lots of entries in it, and it would be hard to find the logs from your last run, and from this specific tool. 

Maybe use https://golang.org/pkg/io/ioutil/#TempDir and prefix it with the name of the executable?

Reply:
works